### PR TITLE
Add Object missing tests

### PR DESCRIPTION
**Pull request by JoseExposito**
Hi!

I submitted part of this changes to my mentee application. I don't think it was accepted (:sob:) but it doesn't matter, I think we can still merge them :wink: 

This PR adds tests that were missing for the `Napi::Object` class. I splitted the changes in commits to make life easier for the code reviewer:

- [Test that Object::GetPropertyNames does not return properties whose key is a Symbol](https://github.com/nodejs/node-addon-api/commit/65d259d45db80a9aecf0e50e7a19532520c5f932)
  As detailed [in the documentation](https://github.com/nodejs/node-addon-api/blob/master/doc/object.md#getpropertynames) when a `Symbol` is used as key, it should not be returned by `Object::GetPropertyNames`.

- [Test that Object::Get returns undefined if the key does not exist](https://github.com/nodejs/node-addon-api/commit/47fcf4a51254e36d20505419bf8661ae17b3e5ce)
  The docs [point out](https://github.com/nodejs/node-addon-api/blob/master/doc/object.md#get) that `Object::Get` must return `undefined` if the key does not exist. I added an extra test for that case and made a quick refactor to avoid repetition.

- [Add tests for Object::InstanceOf](https://github.com/nodejs/node-addon-api/commit/cdd5c9f7495cb7e5713ca62f525cc5cb9810ba6b)

- [Add tests for Object::Object empty constructor](https://github.com/nodejs/node-addon-api/commit/785c38fa52370eb529a08c6a66712e886c73a188)
Here I tested that the returned object to JavaScript is `undefined`. I wasn't sure if this was the right approach or instead I should test that `Value::IsEmpty()`. I'd appreciate your advice here.

- [Add tests for subscript operator](https://github.com/nodejs/node-addon-api/commit/cc65d0aacac72e6a3d84f3823c809b1220867c03)

At this point, the only method [in this table](https://github.com/nodejs/node-addon-api/issues/332) that is not being tested is `Object(napi_env env, napi_value value)`. I didn't implement it because it basically calls `Value(env, value)` so I think it makes sense to implement it in `basic_types/value`, but let me know what you think.

### CODE REVIEW

File: test/object/subscript_operator.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+function test(binding) {
+  function testProperty(obj, key, value, nativeGetProperty, nativeSetProperty) {
+    nativeSetProperty(obj, key, value);
+    assert.strictEqual(nativeGetProperty(obj, key), value);
+  }
+
+  testProperty({}, 'key', 'value', binding.object.subscriptGetWithCStyleString, binding.object.subscriptSetWithCStyleString);
+  testProperty({ key: 'override me' }, 'key', 'value', binding.object.subscriptGetWithCppStyleString, binding.object.subscriptSetWithCppStyleString);
+  testProperty({}, 0, 'value', binding.object.subscriptGetAtIndex, binding.object.subscriptSetAtIndex);
```
Comment (NickNaso) [inline test/object/subscript_operator.js:17]: Could you add:
```js
testProperty({key: 'override me'}, 0, 'value', binding.object.subscriptGetAtIndex, binding.object.subscriptSetAtIndex);
```

**Review by NickNaso — COMMENTED** [review]
@JoseExposito thanks for wirking on this. Just some comments for now.

**Comment by NickNaso** [comment]
>At this point, the only method in this table that is not being tested is Object(napi_env env, napi_value value). I didn't implement it because it basically calls Value(env, value) so I think it makes sense to implement it in basic_types/value, but let me know what you think.

In my opinion you should create a test for `Napi::Object::Object(napi_env env, napi_value value)` here and check if you are able to create a JavaScript object from another JavaScript object.

File: test/object/object.cc [hunk 2]
```diff
@@ -69,6 +77,10 @@ Value TestFunctionWithUserData(const CallbackInfo& info) {
   return Number::New(info.Env(), holder->value);
 }
 
+Value GetEmptyConstructor(const CallbackInfo& /*info*/) {
+  return Object();
```
[the remark this answers is not in the document]
Comment (JoseExposito) [reply test/object/object.cc]: Sure, added in https://github.com/nodejs/node-addon-api/pull/923/commits/4abe5b8e91ef5e97f31868925f83acb6fe05213c

To be consistent with the Function constructor I implemented it as you suggested in the other PR.

File: test/object/subscript_operator.js (hunk 1 above)
Comment (JoseExposito) [reply test/object/subscript_operator.js:17]: Of course, added in https://github.com/nodejs/node-addon-api/pull/923/commits/8e2ee12daedff316f3c0ff1316bae820b7c375cb

**Comment by JoseExposito** [comment]
Hi @NickNaso 

Thanks for you code review

> In my opinion you should create a test for Napi::Object::Object(napi_env env, napi_value value) here and check if you are able to create a JavaScript object from another JavaScript object.

Added in https://github.com/nodejs/node-addon-api/pull/923/commits/61a0c7698d510305d3f1fd7d3116f5e889a8e379

**Review by NickNaso — APPROVED** [review]
LGTM.

**Review by mhdawson — APPROVED** [review]
LGTM

**Review by mhdawson — APPROVED** [review]
LGTM

**Comment by mhdawson** [comment]
In terms of:

**Comment by mhdawson** [comment]
Landed in 298ff8d9d28f32156cb1addc33c2434891697920. @JoseExposito thanks for the PR!
